### PR TITLE
[Class] Enable nullability + fix a few code updates.

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Enable nullability and fix code accordingly.
* Use 'is' and 'is not' instead of '==' and '!=' for object identity.
* Use 'nameof (parameter)' instead of string constants.
* Make the 'handle' field private.
* Remove the (IntPtr) constructor for .NET
* Make the (IntPtr, bool) constructor internal for .NET
* Remove unused 'Construct' method.

### CODE REVIEW

**Review by mandel-macaque — APPROVED** [review]

File: src/ObjCRuntime/Class.cs [hunk 1]
```diff
@@ -66,34 +70,37 @@ public Class (Type type)
...
+#else
 		public Class (IntPtr handle)
+#endif
 		{
 			this.handle = handle;
 		}
 
 		[Preserve (Conditional = true)]
+#if NET
+		internal Class (IntPtr handle, bool owns)
```
Comment (Therzok) [inline src/ObjCRuntime/Class.cs:84]: Is this no longer available as API?

Comment (rolfbjarne) [reply src/ObjCRuntime/Class.cs:80]: @Therzok I ended up putting the other IntPtr constructor back in.

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* monotouch-test/Mac [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2525
* monotouch-test/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2525
* monotouch-test/watchOS 32-bits - simulator/Debug (static registrar): https://github.com/xamarin/maccore/issues/581
* monotouch-test/watchOS 32-bits - simulator/Release (all optimizations): https://github.com/xamarin/maccore/issues/581
* xammac tests/Mac Modern/Debug: https://github.com/xamarin/maccore/issues/2525
* xammac tests/Mac Modern/Release: https://github.com/xamarin/maccore/issues/2525
* xammac tests/Mac Modern/Release (all optimizations): https://github.com/xamarin/maccore/issues/2525

**Pull request merged**